server/src/test/java/org/opensearch/snapshots/RestoreServiceIntegTests.java

```diff
@@ -0,0 +1,241 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.snapshots;
+
+import org.opensearch.action.StepListener;
+import org.opensearch.action.admin.cluster.snapshots.create.CreateSnapshotResponse;
+import org.opensearch.action.admin.cluster.snapshots.restore.RestoreSnapshotRequest;
+import org.opensearch.action.admin.cluster.snapshots.restore.RestoreSnapshotResponse;
+import org.opensearch.action.admin.indices.alias.IndicesAliasesRequest;
+import org.opensearch.action.admin.indices.close.CloseIndexRequest;
+import org.opensearch.action.admin.indices.close.CloseIndexResponse;
+import org.opensearch.action.admin.indices.delete.DeleteIndexRequest;
+import org.opensearch.action.admin.indices.open.OpenIndexRequest;
+import org.opensearch.action.admin.indices.open.OpenIndexResponse;
+import org.opensearch.action.bulk.BulkRequest;
+import org.opensearch.action.bulk.BulkResponse;
+import org.opensearch.action.index.IndexRequest;
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.action.support.WriteRequest;
+import org.opensearch.action.support.master.AcknowledgedResponse;
+import org.opensearch.common.CheckedConsumer;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.repositories.fs.FsRepository;
+import org.opensearch.search.builder.SearchSourceBuilder;
+import org.opensearch.test.OpenSearchIntegTestCase;
+import org.opensearch.test.OpenSearchSingleNodeTestCase;
+
+import java.util.Collections;
+import java.util.Objects;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+public class RestoreServiceIntegTests extends OpenSearchSingleNodeTestCase {
+    // TODO there is certainly a better way to do this, but I don't know what it is....
```

Comment:
A 5-deep nested loop iterating over true/false?  I'm not sure that's better. 😄 

You could iterate an integer and use bitshifting and bitmasking to generate all the combinations?  Which brings up the question, there are 16 possible combinations but not all are used. Which ones aren't?  Might be worth a short comment.

Reply:
I was actually thinking something like parameterized tests, but that doesn't seems to exists in JUnit 3.

Reply:
We should be on Unit 4: https://github.com/opensearch-project/OpenSearch/blob/0f7d572da1d2dec4c2b83c6ae2c9e47ce38edd00/buildSrc/version.properties#L61

And there are other tests that are parameterized:
https://github.com/opensearch-project/OpenSearch/blob/0f7d572da1d2dec4c2b83c6ae2c9e47ce38edd00/server/src/internalClusterTest/java/org/opensearch/search/scroll/SearchScrollWithFailingNodesIT.java#L65-L71

Reply:
Okay - I have adjusted this.